### DISCUSSION TITLE
Local Web UI to Write Scripts

### DISCUSSION THREAD

**Discussion by CameronSchuetz** [crewAIInc/crewAI#1328] [Show and tell]
[CrewUI 1.0.zip](https://github.com/user-attachments/files/17007874/CrewUI.1.0.zip)

A simple UI design to quickly write out and download multi-crew multi-task Python scripts or JSON save files to re-load designs.

**Comment by allina9686** [comment]
First, you can set up a text editor area using a library like Ace Editor. This will provide syntax highlighting and make coding easier. Then, add some action buttons: one for "Save as JSON" so users can download their work as JSON files, another for "Download Script" to save the content as a .py Python script, and a button for "Load Design" to let users upload existing JSON files into the editor.

For the functionality, you can use JavaScript's Blob and URL.createObjectURL() to enable file downloads. To read JSON files, implement a file input and use FileReader to display the contents in the editor.

Here’s a quick code snippet to help you get started:
function downloadFile(content, fileName, contentType) {
    const blob = new Blob([content], { type: contentType });
    const url = URL.createObjectURL(blob);
    const a = document.createElement('a');
    a.href = url;
    a.download = fileName;
    a.click();
}

// Save JSON
document.getElementById('saveJsonBtn').addEventListener('click', () => {
    const jsonContent = document.getElementById('textArea').value;
    downloadFile(jsonContent, 'design.json', 'application/json');
});

// Save Python Script
document.getElementById('saveScriptBtn').addEventListener('click', () => {
    const scriptContent = document.getElementById('textArea').value;
    downloadFile(scriptContent, 'script.py', 'text/x-python');
});
With this setup, users will have an easy time writing, saving, and loading their multi-crew and multi-task scripts in both Python and JSON formats.
